Draft - feat: add support for S3 Extra arguments in Seed file upload to s3

linkerzx opened the pull request:
### Description

<!--- Add a little description on what you plan to tackle with this PR -->
<!--- Add resolve #issue-number in case you resolve an open issue -->

## Models used to test - Optional
<!--- Add here the models that you use to test the changes -->


## Checklist
- [ ] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [ ] You kept your Pull Request small and focused on a single feature or bug fix.
- [ ] You added unit testing when necessary
- [ ] You added functional testing when necessary

nicor88 commented on dbt/include/athena/profile_template.yml:
```diff
@@ -17,6 +17,9 @@ prompts:
     hint: Specify the database (Data catalog) to build models into (lowercase only)
     default: awsdatacatalog
 
+  extra_args:
```
`seed_s3_upload_args`

svdimchenko commented on README.md:
```diff
@@ -88,6 +88,7 @@ A dbt profile can be configured to run against AWS Athena using the following co
 | aws_profile_name      | Profile to use from your AWS shared credentials file.                          | Optional  | `my-profile`                               |
 | work_group            | Identifier of Athena workgroup                                                 | Optional  | `my-custom-workgroup`                      |
 | num_retries           | Number of times to retry a failing query                                       | Optional  | `3`                                        |
+| extra_args | Dictionary containing boto3 ExtraArgs when uploading to S3   | Optional  | `{"ACL": "bucket-owner-full-control"}`|
```
let's apply formatting for markdown table

svdimchenko commented:
overall the code looks good to me. 
The parameter name should be definitely fixed, I like naming which @nicor88 has offered.
Also it would be nice to add `boto3-stubs[s3]~=1.28` row to `dev-requirements.txt`. 

BTW @nicor88 WDYT about migrating from dev-requirements.txt and dependencies in setup.py to poetry ? It may help with dependencies management.

nicor88 commented:
@svdimchenko I think that we discussed that with @Jrmyy back then - we decided to use pip with requirements to adapt a bit with the landscape of other dbt adapters - but I'm happy to reconsider to use it again.

linkerzx commented:
Hi @nicor88 , @svdimchenko - made some changes following the initial review: 
1. Changed the parameter name to seed_s3_upload_args
2. Applied formatting to the markdown tables 
3. Added an example on how to use the arguments in readme
4. Added  boto3-stubs[s3]~=1.28 to dev-requirements
5. Merged on top of current main

svdimchenko commented on dev-requirements.txt:
```diff
@@ -12,3 +12,5 @@ pytest-cov~=4.1
 pytest-xdist~=3.3
 pyupgrade~=3.10
+boto3-stubs[s3]~=1.28
+
```
```suggestion
```
